packages/gatsby-image/src/index.js

```diff
@@ -384,33 +379,49 @@ class Image extends React.Component {
 
             {/* Once the image is visible, start downloading the image */}
             {this.state.isVisible && (
-              <Img
-                alt={alt}
-                title={title}
-                width={image.width}
-                height={image.height}
-                srcSet={image.srcSet}
-                src={image.src}
-                style={imageStyle}
-                onLoad={() => {
-                  this.setState({ imgLoaded: true })
-                  this.props.onLoad && this.props.onLoad()
-                }}
-              />
+              <picture>
+                {image.srcSetWebp && (<source
+                  type={`image/webp`}
+                  srcSet={image.srcSetWebp}
+                  sizes={image.sizes}
+                />)}
+
+                <source
+                  srcSet={image.srcSet}
+                  sizes={image.sizes}
+                />
+
+                <Img
+                  alt={alt}
+                  title={title}
+                  width={image.width}
+                  height={image.height}
+                  src={image.src}
+                  srcSet={image.srcSet}
+                  sizes={image.sizes}
+                  style={imageStyle}
+                  onLoad={() => {
+                    this.setState({ imgLoaded: true })
+                    this.props.onLoad && this.props.onLoad()
+                  }}
+                />
+              </picture>
             )}
 
             {/* Show the original image during server-side rendering if JavaScript is disabled */}
-            <noscript
-              dangerouslySetInnerHTML={{
-                __html: noscriptImg({
-                  alt,
-                  title,
-                  width: image.width,
-                  height: image.height,
-                  ...image,
-                }),
-              }}
-            />
+            {!this.props.critical && (
```

Comment:
May need to walk this back - if `fadeIn` is true - and there is no scripting enabled - the loaded image will not appear

Reply:
this has been revised like so:

https://github.com/gatsbyjs/gatsby/blob/198535f2de1a47fde42577dcf2aed0e86f874867/packages/gatsby-image/src/index.js#L323